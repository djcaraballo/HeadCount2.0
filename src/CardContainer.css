.card-wrapper{
  display: grid;
  grid-template-columns: repeat(auto-fill, 15rem);
  grid-gap: 10px;
}