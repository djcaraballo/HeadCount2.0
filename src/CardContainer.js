import React from 'react';
import PropTypes from 'prop-types';

const CardContainer = (districtData) => {
  return(
    <h2>District</h2>
  )
}

CardContainer.propTypes = {
  districtData: PropTypes.array.isRequired
}

export default CardContainer;