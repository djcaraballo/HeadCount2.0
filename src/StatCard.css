.stat-card {
  -moz-box-shadow: 0 10px 6px -6px #777;
  -webkit-box-shadow: 0 10px 6px -6px #777;
  border: 1px solid #ebebeb;
  border-radius: 1rem;
  box-shadow: 0 10px 6px -6px #777;
  box-sizing: border-box;
  height: 20rem;
  width: 15rem;
}

li {
  list-style-type: none;
  line-height: 1.5em;
  font-size: 1em;
}

.greaterThan {
  color: #038198;
  /*font-weight: bold;*/
}

.lessThan {
  color: #ec03ac;
  font-style: italic;
}

.district-name {
  color: #525252;
}