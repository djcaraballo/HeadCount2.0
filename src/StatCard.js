import React from 'react';
import PropTypes from 'prop-types';

const StatCard = (districtData) => {
  return(
    <div>
      <h2>District</h2>
      
    </div>
  )
}

StatCard.propTypes = {
  districtData: PropTypes.array.isRequired
}

export default StatCard;