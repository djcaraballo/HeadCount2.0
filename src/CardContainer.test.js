import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';

import CardContainer from './CardContainer';
import StatCard from './StatCard';

describe('CardContainer component', () => {

  test('it should match the snapshot', () => {
    const mockdistrictData = [{district: 'Colorado', stats: {2004: 0.123, 2005: 0.456}}, {district: 'Aurora', stats: {2004: 0.789, 2005: 0.101}}]
    const wrapper = shallow(<CardContainer districtData={mockdistrictData} />)

    expect(wrapper).toMatchSnapshot()
  })

  test('it should render all the cards', () => {
    const mockdistrictData = [{district: 'Colorado', stats: {2004: 0.123, 2005: 0.456}}, {district: 'Aurora', stats: {2004: 0.789, 2005: 0.101}}]
    const wrapper = shallow(<CardContainer districtData={mockdistrictData} />)

    expect(wrapper.find(StatCard).length).toEqual(1)
  })
})