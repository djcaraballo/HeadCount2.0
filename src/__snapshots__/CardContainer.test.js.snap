// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer component it should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    districtData={
      Array [
        Object {
          "district": "Colorado",
          "stats": Object {
            "2004": 0.123,
            "2005": 0.456,
          },
        },
        Object {
          "district": "Aurora",
          "stats": Object {
            "2004": 0.789,
            "2005": 0.101,
          },
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          District
        </h2>,
        <StatCard />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "District",
        },
        "ref": null,
        "rendered": "District",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            District
          </h2>,
          <StatCard />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "District",
          },
          "ref": null,
          "rendered": "District",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
